crypto/ed25519/ed25519.go

```diff
@@ -34,6 +45,14 @@ const (
 	SeedSize = 32
 
 	KeyType = "ed25519"
+
+	// cacheSize is the number of public keys that will be cached in
+	// an expanded format for repeated signature verification.
+	//
+	// TODO/perf: Either this should exclude single verification, or be
```

Comment:
We have an artificial limit of 10.000 validators (https://github.com/tendermint/tendermint/blob/HEAD/types/vote_set.go?q=maxvalset#L18). Should we open an issue with this? I don't think this will come up for a while because there are other limitations that would prevent 4096 validators.